### PR TITLE
Added mulVec function for matrices

### PR DESCRIPTION
**Pull request by omprakaash**
Added function to multiply a matrix with a vector (Matrix* Vec).

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/math/mat.zig [hunk 1]
```diff
@@ -636,3 +652,39 @@ test "Mat4x4_perspective" {
...
+        &math.vec3(0, 2, 0),
+        &math.vec3(0, 0, 3),
+    );
+
+    const m = math.Mat3x3.mulVec(&mat, &v);
+    const expected = math.vec3(2,2,3);
+    try testing.expect(math.Vec3, expected).eql(m);
+}
+
+test "Mat4x4_mulVec_ve4" {
```
Comment (slimsag) [inline src/math/mat.zig:678]:
```suggestion
test "Mat4x4_mulVec_vec4" {
```

File: src/math/mat.zig [hunk 2]
```diff
@@ -357,6 +360,19 @@ pub fn Mat(
             return result;
         }
 
+        /// Matrix - Vector multiplicatino: Mat * Vec
```
Comment (slimsag) [inline src/math/mat.zig:363]:
```suggestion
        /// Matrix * Vector multiplication
```

**Review by slimsag — APPROVED** [review]
Thanks for sending this!

**Comment by slimsag** [comment]
@omprakaash would you be interested in adding Vec*Mat (the opposite order)? I think that's all we need to close https://github.com/hexops/mach/issues/1004

**Comment by omprakaash** [comment]
> @omprakaash would you be interested in adding Vec*Mat (the opposite order)? I think that's all we need to close #1004

Sure ! Will get started pretty soon.

**Pull request merged**